framework/src/main/java/org/checkerframework/common/wholeprograminference/scenelib/ASceneWrapper.java

```diff
@@ -0,0 +1,246 @@
+                        method.returnType, TypeUseLocation.RETURN, annosToIgnore);
+                // Receiver type
+                removeIgnoredAnnosFromATypeElement(
+                        method.receiver.type, TypeUseLocation.RECEIVER, annosToIgnore);
+                // Parameter types
+                for (AField param : method.parameters.values()) {
+                    removeIgnoredAnnosFromATypeElement(
+                            param.type, TypeUseLocation.PARAMETER, annosToIgnore);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes all annotations that should be ignored from an ATypeElement.
+     *
+     * @param typeElt the type element from which to remove annotations
+     * @param loc the location where typeElt in used
+     * @param annosToIgnore maps the toString() representation of an ATypeElement and its
+     *     TypeUseLocation to a set of names of annotations that should not be added to .jaif files
+     *     for that location.
+     */
+    private void removeIgnoredAnnosFromATypeElement(
+            ATypeElement typeElt,
+            TypeUseLocation loc,
+            Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        String firstKey = typeElt.description.toString() + typeElt.tlAnnotationsHere;
+        Set<String> annosToIgnoreForLocation = annosToIgnore.get(Pair.of(firstKey, loc));
+        if (annosToIgnoreForLocation != null) {
+            Set<Annotation> annosToRemove = new HashSet<>();
+            for (Annotation anno : typeElt.tlAnnotationsHere) {
+                if (annosToIgnoreForLocation.contains(anno.def().toString())) {
+                    annosToRemove.add(anno);
+                }
+            }
+            typeElt.tlAnnotationsHere.removeAll(annosToRemove);
+        }
+
+        // Recursively remove ignored annotations from inner types
+        if (!typeElt.innerTypes.isEmpty()) {
+            for (ATypeElement innerType : typeElt.innerTypes.values()) {
+                removeIgnoredAnnosFromATypeElement(innerType, loc, annosToIgnore);
+            }
+        }
+    }
+
+    /**
+     * Write the scene wrapped by this object to a jaif on the given path.
+     *
+     * @param jaifPath the path to the jaif file to be written
+     * @param annosToIgnore which annotations should be ignored in which contexts
+     */
+    public void writeToJaif(
+            String jaifPath, Map<Pair<String, TypeUseLocation>, Set<String>> annosToIgnore) {
+        try {
+            AScene scene = theScene.clone();
+            removeIgnoredAnnosFromScene(scene, annosToIgnore);
+            scene.prune();
+            new File(jaifPath).delete();
+            if (!scene.isEmpty()) {
+                // Only write non-empty scenes into .jaif files.
+                IndexFileWriter.write(scene, new FileWriter(jaifPath));
+            }
+        } catch (IOException e) {
+            throw new UserError(
+                    "Problem while reading file in: "
```

Comment:
Isn't the problem while writing?  The code for this in `writeToStub` is much better and should probably be adopted here.